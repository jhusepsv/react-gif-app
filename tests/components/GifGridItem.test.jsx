import { render, screen } from "@testing-library/react"
import { GifGridItem } from "../../src/components/GifGridItem"

describe('Pruebas en el nombre del componente', () => { 

   const title = 'Saitama';
   const url = 'https://one-punch.com/saitama.jpg'

   test('debe hacer march con el snapshot', () => { 
      const { container } = render(<GifGridItem title={title} url={url}/>);
      expect(container).toMatchSnapshot();
   });

   test('debe mostrar la imagen con el URL y el ALT indicado', () => { 
      render(<GifGridItem title={title} url={url}/>);
      // screen.debug();
      // expect( screen.getByRole( 'img' ).src ).toBe( url );
      const { src, alt } = screen.getByRole('img');
      expect(src).toBe(url);
      expect(alt).toBe(title);
   });

   test('debe mostrar el titulo en el componente', () => { 
      render(<GifGridItem title={title} url={url}/>);
      expect( screen.getByText(title) ).toBeTruthy();
   });

});
