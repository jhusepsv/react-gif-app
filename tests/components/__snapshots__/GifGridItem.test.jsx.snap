// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el nombre del componente debe hacer march con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
